src/components/ui/Drawer.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * This file is based on the Sheet component from shadcn and customized for our needs.
+ * The Sheet component of shadcn is based on the Dialog component of Radix UI.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/sheet
+ * Radix UI: https://www.radix-ui.com/themes/docs/components/dialog
+ */
+'use client';
+
+import { Icon } from '@/components/ui';
+import { cn } from '@/lib/tailwind/utils';
+
+import {
+  Close as PrimitiveClose,
+  Content as PrimitiveContent,
+  Description as PrimitiveDescription,
+  Overlay as PrimitiveOverlay,
+  Portal as PrimitivePortal,
+  Root as PrimitiveRoot,
+  Title as PrimitiveTitle,
+  Trigger as PrimitiveTrigger,
+} from '@radix-ui/react-dialog';
+import { cva, type VariantProps } from 'class-variance-authority';
+import { ComponentPropsWithoutRef, ElementRef, forwardRef, HTMLAttributes } from 'react';
+
+const DrawerRoot = PrimitiveRoot;
+const DrawerTrigger = PrimitiveTrigger;
+const DrawerClose = PrimitiveClose;
+const DrawerPortal = PrimitivePortal;
+
+const DrawerOverlay = forwardRef<
+  ElementRef<typeof PrimitiveOverlay>,
+  ComponentPropsWithoutRef<typeof PrimitiveOverlay>
+>(({ className, ...props }, ref) => (
+  <PrimitiveOverlay
+    className={cn(
```

Comment:
I liked how you classify className by it's categories!
Maybe we can create a note or doc somewhere about how we classify and organize classNames in cn() method

Reply:
I was considering creating a `STYLEGUIDE.md` file to document coding rules for a project, such as using `I` to name interfaces like `IMyProps`.

Reply:
Created #91